src/PostgREST/DbStructure/Relation.hs

```diff
@@ -32,45 +29,36 @@ data Relation = Relation
   , relColumns  :: [Column]
   , relFTable   :: Table
   , relFColumns :: [Column]
-  , relType     :: Cardinality
-  , relLink     :: Link -- ^ Constraint on O2M/M2O, Junction for M2M Cardinality
+  , relCard     :: Cardinality
   }
   deriving (Eq, Generic, JSON.ToJSON)
 
+-- | The relationship cardinality
+-- | https://en.wikipedia.org/wiki/Cardinality_(data_modeling)
+-- TODO: missing one-to-one
+data Cardinality
+  = O2M ConstraintName -- ^ one-to-many cardinality
+  | M2O ConstraintName -- ^ many-to-one cardinality
+  | M2M Junction       -- ^ many-to-many cardinality
+  deriving (Eq, Generic, JSON.ToJSON)
+
 type ConstraintName = Text
```

Comment:
One more thing about naming came to mind: Now, that we don't have `Constraint` anymore, why don't we just:

```suggestion
type Constraint = Text
```

Reply:
Going to rename it ot `FKConstraint`, seems clearer. Also, otherwise:

```
    Ambiguous occurrence ‘Constraint’
    It could refer to
       either ‘Protolude.Constraint’,
```